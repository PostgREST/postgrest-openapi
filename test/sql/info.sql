-- shows the first line of the comment on the schema as title
select jsonb_pretty(postgrest_openapi_spec('{test}')->'info'->'title');

-- shows the second line of the comment on the schema as description
select jsonb_pretty(postgrest_openapi_spec('{test}')->'info'->'description');

-- TODO: tests for versions
