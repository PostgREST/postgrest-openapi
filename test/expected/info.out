-- shows the first line of the comment on the schema as title
select jsonb_pretty(postgrest_openapi_spec('{test}')->'info'->'title');
  jsonb_pretty  
----------------
 "My API title"
(1 row)

-- shows the second line of the comment on the schema as summary
select jsonb_pretty(postgrest_openapi_spec('{test}')->'info'->'summary');
   jsonb_pretty   
------------------
 "My API summary"
(1 row)

-- shows the third line of the comment on the schema as description
select jsonb_pretty(postgrest_openapi_spec('{test}')->'info'->'description');
                   jsonb_pretty                   
--------------------------------------------------
 "My API description\nthat spans\nmultiple lines"
(1 row)

-- shows default title when there is no comment on the schema
select jsonb_pretty(postgrest_openapi_spec('{types}')->'info'->'title');
  jsonb_pretty   
-----------------
 "PostgREST API"
(1 row)

-- shows no summary when there is no comment on the schema
select postgrest_openapi_spec('{types}')->'info' ? 'summary';
 ?column? 
----------
 f
(1 row)

  
-- shows default description when there is no comment on the schema
select jsonb_pretty(postgrest_openapi_spec('{types}')->'info'->'description');
                  jsonb_pretty                  
------------------------------------------------
 "This is a dynamic API generated by PostgREST"
(1 row)

-- TODO: tests for versions
